src/Meziantou.AspNetCore.Components/GenericFormField.cs

```diff
@@ -253,6 +253,24 @@ private static (Type ComponentType, IEnumerable<KeyValuePair<string, object>>? A
             if (property.PropertyType == typeof(decimal))
                 return (typeof(InputNumber<decimal>), null);
 
+            if (property.PropertyType == typeof(short?))
```

Comment:
I think it would be better to get the underlying type using `Nullable.GetUnderlyingType(property.PropertyType)` and use this value for the existing `if`. In the `return` statements, you can create the actual type using `typeof(InputNumber<>).MakeGenericType(property.PropertyType)`. This would prevent the duplication of `if`

Reply:
I've made the change in 4cd7a4a0e001ea6eaeb480cae953ea0bb2bea65b (I added you as a co-author of the commit 😊)